### PR TITLE
Added current location

### PR DESCRIPTION
**Pull request by shra-wtag**
Added current location and current location button

### CODE REVIEW

File: WeatherApp/Model/CityModel.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+struct CityModel {
+    let name: String
+    let latitude: Double
+    let longitude: Double
+}
+
+
```
Comment (lif-wtag) [inline WeatherApp/Model/CityModel.swift]: Remove Empty Line

File: WeatherApp/View/CustomLocationCell/LocationTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @IBOutlet weak var locationCustomCell: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
```
Comment (lif-wtag) [inline WeatherApp/View/CustomLocationCell/LocationTableViewCell.swift]: Remove commented lines

File: WeatherApp/ViewControllers/SearchCityViewController.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        registerCustomLocationCell()
+        searchData = locationList ?? []
+    }
+    
+    func registerCustomLocationCell() {
+        let nib = UINib(nibName: "LocationTableViewCell", bundle: nil)
+        locationTableView.register(nib, forCellReuseIdentifier: "locationTableViewCell")
+    }
+    
+    func allCities() {
```
Comment (lif-wtag) [inline WeatherApp/ViewControllers/SearchCityViewController.swift]: rename this function prepareCityList()

**Review by lif-wtag — CHANGES_REQUESTED** [review]

File: WeatherApp/Services/NetworkManager.swift [hunk 4]
```diff
@@ -1,19 +1,24 @@
 import Foundation
 
 class NetworkManager {
-    
-    func fetchFromApi(completionHandler:  @escaping (WeatherInfoData) -> Void) {
+    func fetchWeatherData(latitude: Double, longitude: Double, completionHandler:  @escaping (WeatherInfoData?) -> Void) {
         DispatchQueue.global(qos: .userInitiated).async {
-            let url = URL(string: "https://api.openweathermap.org/data/2.5/weather?lat=23.810331&lon=90.412521&units=metric&appid=f7fba2a96004431c6e3b90fb0728bd89")
-            let dataTask = URLSession.shared.dataTask(with: url!) { (data, response, error) in
+            let url = URL(string: "https://api.openweathermap.org/data/2.5/weather?lat=\(latitude)&lon=\(longitude)&units=metric&appid=f7fba2a96004431c6e3b90fb0728bd89")
```
Comment (ash-wtag) [inline WeatherApp/Services/NetworkManager.swift]: Please consider using constant file to store the base url, appid

File: WeatherApp/Services/NetworkManager.swift [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
+            let dataTask = URLSession.shared.dataTask(with: url!) { (data, response, error)  in
                 guard let data = data, error == nil else {
                     print("Error : \(error!.localizedDescription)")
+                    completionHandler(nil)
                     return
                 }
                 var fetchedWeatherData: WeatherInfoData?
```
Comment (ash-wtag) [inline WeatherApp/Services/NetworkManager.swift]: Is there any reason to declare this variable here? you can consider declare it as let inside do scope, as you're not changing it later on

File: WeatherApp/ViewControllers/SearchCityViewController.swift [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        searchData = []
+    
+        guard let searchInput = searchCity.text, let locationList = locationList else {
+            return
+        }
+        if searchInput.isEmpty {
+            searchData = locationList
+        }
+        else {
```
Comment (ash-wtag) [inline WeatherApp/ViewControllers/SearchCityViewController.swift:51]: Please add proper indentation

File: WeatherApp/ViewControllers/SearchCityViewController.swift [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            searchData = locationList.filter({
+                $0.name.lowercased().contains(searchText.lowercased())
```
Comment (ash-wtag) [inline WeatherApp/ViewControllers/SearchCityViewController.swift:53]: Do you have plan to integrate mapbox auto complete places? or it just filter the local city list>

File: WeatherApp/ViewControllers/WeatherInfoViewController.swift [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+                self.updateWeatherDataInView()
+            }
+        })
+    }
+    
+    func updateWeatherDataInView() {
+        if let weatherInfoData = weatherInfoData {
+            cityName.text = "\(weatherInfoData.name)"
+            self.temperatureLabel.text = "\(Int(weatherInfoData.main.temp))ºC"
+            self.descriptionLabel.text = "\(weatherInfoData.weather[0].description)"
```
Comment (ash-wtag) [inline WeatherApp/ViewControllers/WeatherInfoViewController.swift]: Please add a condition to check if the list is empty or not

File: WeatherApp/ViewControllers/WeatherInfoViewController.swift [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+            let iconName = weatherInfoData.weather[0].icon
```
Comment (ash-wtag) [inline WeatherApp/ViewControllers/WeatherInfoViewController.swift]: ditto

File: WeatherApp/ViewControllers/WeatherInfoViewController.swift [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+            self.descriptionIcon.image = UIImage(named: iconName)
+            self.feelsLikeLabel.text = "Feels like \(Int(weatherInfoData.main.feels_like))ºC"
```
Comment (ash-wtag) [inline WeatherApp/ViewControllers/WeatherInfoViewController.swift]: Any reason to convert it to int?

Comment (shra-wtag) [reply WeatherApp/ViewControllers/WeatherInfoViewController.swift]: > Any reason to convert it to int?

The api's temperature gives a double value, so to get an integer value I converted it to int.

File: WeatherApp/Resources/Assets.xcassets/09d.imageset/09d.png
Comment (ash-wtag) [inline WeatherApp/Resources/Assets.xcassets/09d.imageset/09d.png:1]: please add proper name to assets files

File: WeatherApp/ViewControllers/SearchCityViewController.swift (hunk 7 above)
Comment (shra-wtag) [reply WeatherApp/ViewControllers/SearchCityViewController.swift:53]: For now, I've created instances of the CityModel structure to filter from those instances. I'll do further updates in future.

File: WeatherApp/AppAndSceneDelegate/AppDelegate.swift [hunk 11]
```diff
@@ -1,4 +1,5 @@
 import UIKit
+import CoreData
```
Comment (fka-wtag) [inline WeatherApp/AppAndSceneDelegate/AppDelegate.swift]: why do you need it here?

Comment (shra-wtag) [reply WeatherApp/AppAndSceneDelegate/AppDelegate.swift]: I'll remove it, it was kept by mistake.

**Pull request merged**